### PR TITLE
Add SparkFun Thing Plus RP2350

### PR DESCRIPTION
**Pull request by sfe-SparkFro**
Copied parts from SparkFun Pro Micro RP2350 and Pico W. Tested on a board and all seems well.

### CODE REVIEW

**Comment by jepler** [comment]
Just checking -- This board has wifi with cyw43?

File: ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+static const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_D1), MP_ROM_PTR(&pin_GPIO1) },
+    { MP_ROM_QSTR(MP_QSTR_D2), MP_ROM_PTR(&pin_GPIO2) },
+    { MP_ROM_QSTR(MP_QSTR_D3), MP_ROM_PTR(&pin_GPIO3) },
+    { MP_ROM_QSTR(MP_QSTR_D4), MP_ROM_PTR(&pin_GPIO4) },
+    { MP_ROM_QSTR(MP_QSTR_D5), MP_ROM_PTR(&pin_GPIO5) },
```
Comment (jepler) [inline ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c:17]: I don't have the Thing Plus RP2350 to check, but I compared to the Thing Plus RP2040 and saw there's no "D5" pin broken out.

In general, if a pin is not used, the corresponding entry in the board pins file will be omitted.

File: ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+// SPDX-FileCopyrightText: Copyright (c) 2024 Scott Shawcroft for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+
+#include "shared-bindings/board/__init__.h"
+
+static const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO0) },
```
Comment (jepler) [inline ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c:12]: We prefer to group aliases of the same pin together:
```
    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO0) },
    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO0) },
```
(the name that comes FIRST in this file is the one shown when printing the pin.

**Review by jepler — CHANGES_REQUESTED** [review]
Hi! I have a few style comments on the pins but overall this seems in good shape.

File: ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/mpconfigboard.mk
Comment (eightycc) [inline ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/mpconfigboard.mk:1]: The SDK currently uses a default PIO clock divider of 2. Given the higher clock speed of the RP2350 consider increasing it to 3 to reduce the emulated gSPI clock frequency.
```
CFLAGS += -DCYW43_PIO_CLOCK_DIV_INT=3
```

Comment (sfe-SparkFro) [reply ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/mpconfigboard.mk:1]: In my testing, I did not experience any problems with the default clock divider of 2. The Pico SDK also uses the default for the [Pico 2 W](https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pico2_w.h), which I would take to mean the default value is fine for the RP2350. Increasing the clock divider would come at the cost of lower transmit/receive speed, so would prefer to leave it as is unless there's compelling reason to change it.

File: ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c (hunk 1 above)
Comment (sfe-SparkFro) [reply ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c:17]: There's some minor pinout differences between the boards, and GPIO 5 is now broken out. I've double checked that the pins in this file match the pins available to the user. The only pins that are not available are GPIO 8 for the PSRAM, and GPIO 23/24/25/29 for the radio.

File: ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c (hunk 2 above)
Comment (sfe-SparkFro) [reply ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c:12]: Ah! I was unaware of that, good to know! Will get that updated, thanks!

Comment (sfe-SparkFro) [reply ports/raspberrypi/boards/sparkfun_thing_plus_rp2350/pins.c:13]: Ok, how's it look now?

**Review by jepler — COMMENTED** [review]
Thank you, the pin stuff looks good now. Thanks for the other clarifications as well.

As for the divisor issue, I do not see this setting being made in the pico2_w board specific files in the "develop" branch of pico-sdk:
https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pico2_w.h
i.e., they appear to be using the default divisor of 2 as well.

**Review by jepler — APPROVED** [review]

**Pull request merged**